apps/dashboard/src/utils/schema.ts

```diff
@@ -110,63 +111,70 @@ const handleNumberType = ({
   return numberValue;
 };
 
+const getZodValueByType = (jsonSchema: JSONSchemaDefinition, key: string): ZodValue => {
+  if (typeof jsonSchema !== 'object') {
+    return z.any();
+  }
+  const requiredFields = jsonSchema.required ?? [];
+  const { type, format, pattern, enum: enumValues, default: defaultValue, required, minimum, maximum } = jsonSchema;
+
+  if (type === 'object') {
+    let zodValue = buildDynamicZodSchema(jsonSchema, key) as z.ZodTypeAny;
+    if (defaultValue) {
+      zodValue = zodValue.default(defaultValue as object);
+    }
+    zodValue = zodValue.transform((val) => {
+      const hasAnyRequiredEmpty = required?.some((field) => val[field] === '' || val[field] === undefined);
+      // remove object if any required field is empty or undefined
+      return hasAnyRequiredEmpty ? undefined : val;
+    });
+    return zodValue.nullable();
+  } else if (type === 'string') {
+    return handleStringType({ key, requiredFields, format, pattern, enumValues, defaultValue });
+  } else if (type === 'boolean') {
+    return z.boolean();
+  } else if (type === 'number') {
+    return handleNumberType({ key, minimum, maximum, defaultValue, requiredFields });
+  } else if (typeof jsonSchema === 'object' && jsonSchema.anyOf) {
+    const anyOf = jsonSchema.anyOf.map((oneOfObj) => buildDynamicZodSchema(oneOfObj, key));
+
+    return z.union(anyOf as any);
+  } else {
+    return z.any();
+  }
+};
+
 /**
  * Transform JSONSchema to Zod schema.
  * The function will recursively build the schema based on the JSONSchema object.
  * It removes empty strings and objects with empty required fields during the transformation phase after parsing.
  */
-export const buildDynamicZodSchema = (obj: JSONSchemaDto): z.AnyZodObject => {
-  const properties = typeof obj === 'object' ? (obj.properties ?? {}) : {};
-  const requiredFields = typeof obj === 'object' ? (obj.required ?? []) : [];
+export const buildDynamicZodSchema = (obj: JSONSchemaDefinition, key = ''): ZodValue => {
```

Comment:
what was changed here:
- the body of this function was extracted to the `getZodValueByType` 
- handling of `jsonSchema.anyOf` field
- handling simple json schema types like string or number